blueocean-dashboard/src/main/js/components/PipelinePage.jsx

```diff
@@ -11,24 +16,37 @@ import {
     TabLink,
     WeatherIcon,
 } from '@jenkins-cd/design-language';
+import PageLoading from './PageLoading';
 import { buildOrganizationUrl, buildPipelineUrl } from '../util/UrlUtils';
 
-export default class PipelinePage extends Component {
+export class PipelinePage extends Component {
+    getChildContext() {
+        return {
+            pipeline: this.props.pipeline,
+        };
+    }
+    
+    componentWillMount() {
+        if (this.props.params) {
+            this.props.fetchPipeline(this.props.params.organization, this.props.params.pipeline);
+        }
+    }
+    
+    _componentWillReceiveProps(nextProps) {
```

Comment:
+1 on -1 for `_componentWillReceiveProps` ... don't name things **nearly the same** as something else that's very fundamental.